File: api/src/main/java/com/hva/nl/ewa/models/Invite.java
package com.hva.nl.ewa.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

@Entity
@Table(name = "invites")
public class Invite implements Model {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NotNull
    private long inviteId;

    @JsonIgnore
    @OneToOne(targetEntity = User.class, fetch = FetchType.LAZY)
    @JoinColumn(name = "invitee_id")
    @NotNull
    private User invitee;

    @JsonIgnore
    @OneToOne(targetEntity = User.class, fetch = FetchType.LAZY)
    @JoinColumn(name = "inviter_id")
    @NotNull
    private User inviter;

    @JsonIgnore
    @ManyToOne(targetEntity = Game.class, fetch = FetchType.LAZY)
    @JoinColumn(name = "game_id")
    @NotNull
    private Game game;

    public long getInviteId() {
        return inviteId;
    }

    public void setInviteId(long inviteId) {
        this.inviteId = inviteId;
    }

    public User getInvitee() {
        return invitee;
    }

    public void setInvitee(User invitee) {
        this.invitee = invitee;
    }

    public User getInviter() {
        return inviter;
    }

    public void setInviter(User inviter) {
        this.inviter = inviter;
    }

    public Game getGame() {
        return game;
    }

    public void setGame(Game game) {
        this.game = game;
    }
}

File: web/src/app/model/Invite.ts
import {Game} from './Game';

export default class Invite {

  private readonly _inviteId: number;

  private readonly _inviterName: string;

  private readonly _game: Game;

  constructor(inviteId: number, inviterName: string, game: Game) {
    this._inviteId = inviteId;
    this._inviterName = inviterName;
    this._game = game;
  }

  get inviteId(): number {
    return this._inviteId;
  }

  get inviterName(): string {
    return this._inviterName;
  }

  get game(): Game {
    return this._game;
  }
}
